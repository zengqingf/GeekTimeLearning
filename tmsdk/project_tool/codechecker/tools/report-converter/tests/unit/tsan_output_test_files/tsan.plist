<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>diagnostics</key>
	<array>
		<dict>
			<key>category</key>
			<string>unknown</string>
			<key>check_name</key>
			<string>ThreadSanitizer</string>
			<key>description</key>
			<string>SEGV on unknown address 0x000000000000 (pc 0x0000004b525c bp 0x7fff93b54920 sp 0x7fff93b548b0 T23755)</string>
			<key>issue_hash_content_of_line_in_context</key>
			<string>4c95a99bdc3a7ca6f49bfac68164842c</string>
			<key>location</key>
			<dict>
				<key>col</key>
				<integer>2</integer>
				<key>file</key>
				<integer>0</integer>
				<key>line</key>
				<integer>24</integer>
			</dict>
			<key>notes</key>
			<array>
				<dict>
					<key>depth</key>
					<integer>0</integer>
					<key>kind</key>
					<string>note</string>
					<key>location</key>
					<dict>
						<key>col</key>
						<integer>2</integer>
						<key>file</key>
						<integer>0</integer>
						<key>line</key>
						<integer>24</integer>
					</dict>
					<key>message</key>
					<string>==23755==The signal is caused by a WRITE memory access.
==23755==Hint: address points to the zero page.
    #0 insert_in_table(unsigned long, int) files/tsan.cpp:18:14 (a.out+0x4b525b)
    #1 main files/tsan.cpp:24:2 (a.out+0x4b529e)
    #2 __libc_start_main /build/glibc-OTsEL5/glibc-2.27/csu/../csu/libc-start.c:310 (libc.so.6+0x21b96)
    #3 _start &lt;null&gt; (a.out+0x41c8d9)
</string>
				</dict>
			</array>
			<key>path</key>
			<array>
				<dict>
					<key>depth</key>
					<integer>0</integer>
					<key>kind</key>
					<string>event</string>
					<key>location</key>
					<dict>
						<key>col</key>
						<integer>2</integer>
						<key>file</key>
						<integer>0</integer>
						<key>line</key>
						<integer>24</integer>
					</dict>
					<key>message</key>
					<string>    #1 main files/tsan.cpp:24:2 (a.out+0x4b529e)</string>
				</dict>
				<dict>
					<key>depth</key>
					<integer>0</integer>
					<key>kind</key>
					<string>event</string>
					<key>location</key>
					<dict>
						<key>col</key>
						<integer>14</integer>
						<key>file</key>
						<integer>0</integer>
						<key>line</key>
						<integer>18</integer>
					</dict>
					<key>message</key>
					<string>    #0 insert_in_table(unsigned long, int) files/tsan.cpp:18:14 (a.out+0x4b525b)</string>
				</dict>
				<dict>
					<key>edges</key>
					<array>
						<dict>
							<key>end</key>
							<array>
								<dict>
									<key>col</key>
									<integer>14</integer>
									<key>file</key>
									<integer>0</integer>
									<key>line</key>
									<integer>18</integer>
								</dict>
								<dict>
									<key>col</key>
									<integer>14</integer>
									<key>file</key>
									<integer>0</integer>
									<key>line</key>
									<integer>18</integer>
								</dict>
							</array>
							<key>start</key>
							<array>
								<dict>
									<key>col</key>
									<integer>2</integer>
									<key>file</key>
									<integer>0</integer>
									<key>line</key>
									<integer>24</integer>
								</dict>
								<dict>
									<key>col</key>
									<integer>2</integer>
									<key>file</key>
									<integer>0</integer>
									<key>line</key>
									<integer>24</integer>
								</dict>
							</array>
						</dict>
					</array>
					<key>kind</key>
					<string>control</string>
				</dict>
				<dict>
					<key>depth</key>
					<integer>0</integer>
					<key>kind</key>
					<string>event</string>
					<key>location</key>
					<dict>
						<key>col</key>
						<integer>2</integer>
						<key>file</key>
						<integer>0</integer>
						<key>line</key>
						<integer>24</integer>
					</dict>
					<key>message</key>
					<string>SEGV on unknown address 0x000000000000 (pc 0x0000004b525c bp 0x7fff93b54920 sp 0x7fff93b548b0 T23755)</string>
				</dict>
			</array>
			<key>type</key>
			<string>tsan</string>
		</dict>
	</array>
	<key>files</key>
	<array>
		<string>files/tsan.cpp</string>
	</array>
	<key>metadata</key>
	<dict>
		<key>analyzer</key>
		<dict>
			<key>name</key>
			<string>tsan</string>
		</dict>
		<key>generated_by</key>
		<dict>
			<key>name</key>
			<string>report-converter</string>
			<key>version</key>
			<string>x.y.z</string>
		</dict>
	</dict>
</dict>
</plist>
