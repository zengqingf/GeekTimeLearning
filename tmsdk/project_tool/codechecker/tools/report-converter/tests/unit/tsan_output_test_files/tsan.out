ThreadSanitizer:DEADLYSIGNAL
==23755==ERROR: ThreadSanitizer: SEGV on unknown address 0x000000000000 (pc 0x0000004b525c bp 0x7fff93b54920 sp 0x7fff93b548b0 T23755)
==23755==The signal is caused by a WRITE memory access.
==23755==Hint: address points to the zero page.
    #0 insert_in_table(unsigned long, int) files/tsan.cpp:18:14 (a.out+0x4b525b)
    #1 main files/tsan.cpp:24:2 (a.out+0x4b529e)
    #2 __libc_start_main /build/glibc-OTsEL5/glibc-2.27/csu/../csu/libc-start.c:310 (libc.so.6+0x21b96)
    #3 _start <null> (a.out+0x41c8d9)

ThreadSanitizer can not provide additional info.
SUMMARY: ThreadSanitizer: SEGV files/tsan.cpp:18:14 in insert_in_table(unsigned long, int)
==23755==ABORTING